web/packages/shared/components/UnifiedResources/shared/predicateExpression.test.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ResourceStatus } from '../types';
+import { buildPredicateExpression } from './predicateExpression';
+
+describe('getPredicateExpression', () => {
+  const testCases: {
+    name: string;
+    requestedStatuses?: ResourceStatus[];
+    requestedQuery?: string;
+    expected: string | undefined;
+  }[] = [
+    {
+      name: 'undefined fields',
+      requestedStatuses: undefined,
+      requestedQuery: undefined,
+      expected: '',
+    },
+    {
+      name: 'empty status array',
+      requestedStatuses: [],
+      expected: '',
+    },
+    {
+      name: 'empty value in status array',
+      requestedStatuses: ['', ''],
+      expected: '',
+    },
+    {
+      name: 'with only status',
+      requestedStatuses: ['healthy'],
+      expected: 'health.status == "healthy"',
+    },
+    {
+      name: 'with multiple statuses',
+      requestedStatuses: ['healthy', '', 'unhealthy'],
+      expected: 'health.status == "healthy" || health.status == "unhealthy"',
+    },
+    {
+      name: 'with only query',
+      requestedQuery: 'name == "mysql"',
+      expected: 'name == "mysql"',
+    },
+    {
+      name: 'with status and query',
+      requestedStatuses: ['healthy'],
+      requestedQuery: 'name == "mysql"',
+      expected: 'name == "mysql" && (health.status == "healthy")',
+    },
+    {
+      name: 'with multi status and query',
+      requestedStatuses: ['healthy', '', 'unhealthy'],
+      requestedQuery: 'name == "mysql"',
+      expected:
+        'name == "mysql" && (health.status == "healthy" || health.status == "unhealthy")',
```

Comment:
Don't we need to wrap `requestedPredicateExpression` in parentheses too?
What if `requestedPredicateExpression` is `'name == "mysql"' || name == "postgres"`?

The result of:
`'name == "mysql" || name == "postgres" && (health.status == "healthy" || health.status == "unhealthy")',`
is different from:
`('name == "mysql" || name == "postgres") && (health.status == "healthy" || health.status == "unhealthy")',`

Reply:
yes great catch 👍